nimbus-jose-jwt_aws-kms-extension/src/test/java/com/nimbusds/jose/aws/kms/crypto/KmsSymmetricDecrypterTest.java

```diff
@@ -191,73 +175,9 @@ void beforeEach() {
                         testJweHeader);
             }
 
-            @Nested
-            @DisplayName("with invalid key exception from KMS,")
-            class WithInvalidKMSKeyException {
```

Comment:
Since you are removing this test-cases, please add replacement tests by mocking the 'JWEDecrypterUtil'.

Reply:
As I have moved the exceptions to util class and hence added related tests there.

Reply:
For the test moved to the util class we should do one of the following:

1. Mock the util class and write success and error tests (by return a response or throwing an error from the mocked util class).
2. Don't mock the util class and write test-cases covering all the branches in the util class.

Since you have written separate test-cases for the util class, option 1 seems to be the preferred option here. Kindly update the test-cases accordingly.

Reply:
Updated

Reply:
Added a follow-up comment here: https://github.com/amzn/nimbus-jose-jwt_aws-kms-extension/pull/13#discussion_r1540425992

Reply:
Updated